yt_dlp/extractor/bandcamp.py

```diff
@@ -411,7 +410,7 @@ def _real_extract(self, url):
 
 class BandcampWeeklyIE(BandcampIE):  # XXX: Do not subclass from concrete IE
     IE_NAME = 'Bandcamp:weekly'
-    _VALID_URL = r'https?://(?:www\.)?bandcamp\.com/?\?(?:.*?&)?show=(?P<id>\d+)'
+    _VALID_URL = r'https?://(?:www\.)?bandcamp\.com(?:/radio)?/?\?.*?(?<=[?&])show=(?P<id>\d+)'
     _TESTS = [{
         'url': 'https://bandcamp.com/?show=224',
```

Comment:
```suggestion
        'url': 'https://bandcamp.com/radio?show=224',
```

and you can change the `only_matching` test URL below to:
```py
        'url': 'https://bandcamp.com/radio/?foo=bar&show=224',
```

please actually run the test, as well